lib/ProductOpener/Store.pm

```diff
@@ -97,17 +100,37 @@ sub get_string_id_for_lang {
 		}
 	}
 
-	if (not defined $string) {
+	return _sanitize_string($string, $lowercase, $unaccent);
+
+}
+
+sub get_fileid {
+
+	my ($file, $unaccent, $lc) = @_;
+
+	if (not defined $file) {
 		return "";
 	}
 
+	if ((defined $lc) and ($lc eq 'fr')) {
+		$unaccent = 1;
+	}
+
+	return _sanitize_string($file, 1, $unaccent);
+
+}
+
+sub _sanitize_string {
```

Comment:
Given the huge amount of time this function is called, I think it could be worth to inline it in the 2 places it is used. It will double a small amount of code, but we'll save the overhead of calling one more function, copying arguments etc.

Reply:
Is calling `sub`s that expensive in Perl? I know that `shift`ing the array is not the fastest way to access parameters, which is why I changed it to `my ($string, $lowercase, $unaccent) = @_;`, but I'm a bit surprised that an additional method/function is slow, since it's quite uncommon in other languages (ie. Java, C#, JavaScript might even inline short methods during JIT).

Reply:
I've tried to read up on this, and apparently, Perl is actually pretty horrible with each additional sub call. I'll re-inline the method.

- [why are function calls in Perl loops so slow?](https://stackoverflow.com/a/3259215/11963)
- [Performance issues with subroutines/methods](https://www.perlmonks.org/bare/?node_id=621800)
- [Subroutine overhead in Perl](https://www.perlmonks.org/?node_id=649281)
- [Perl Performance - How to make Perl faster](http://makepp.sourceforge.net/2.0/perl_performance.html#functions)